type.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Type System
+---
+
+Common Lisp has a complete and flexible type system and corresponding tools to
+inspect, check and manipulate types.
+
+## Values Have Types, Not Variables
+
+Being different from some languages such as C/C++, variables in Lisp are just
+*placeholders* for values. When you [`SETF`][setf] something, the value is only
+*bound* to the variable. You can bind another value to the same variable later,
+as you wish.
+
+This implies a fact that in Common Lisp **values have types**, while variables
+do not. This might be surprising at first if you come from a C/C++ background.
+
+For example:
+
+~~~lisp
+* (defvar *var* 1234)
+*VAR*
+
+* (type-of *var*)
+(INTEGER 0 4611686018427387903)
+~~~
```

Comment:
To be honest, I think it is "values have types".

Because `(type-of *var*)` will evaluate `*var*` first, then calculate the type of evaluated result. So it in fact returns the type of a *value* instead of `*var*` itself